extensions/WSLExtension/DistributionDefinitions/DistributionDefinitionHelper.cs

```diff
@@ -0,0 +1,115 @@
+            var webDefinitions = JsonSerializer.Deserialize<DistributionDefinitions>(distributionDefinitionsJson, _jsonOptions);
+            var distributionDefinitionsMap = new Dictionary<string, DistributionDefinition>();
+
+            foreach (var definition in webDefinitions!.Values)
+            {
+                // Only supported distributions for this machine.
+                if (ShouldAddDistribution(definition))
+                {
+                    distributionDefinitionsMap.Add(definition.Name, definition);
+                }
+            }
+
+            // Merge the local distribution information we have stored in DistributionDefinition.yaml with the one above.
+            var uri = new Uri(KnownDistributionsLocalYamlLocation);
+            var storageFile = await StorageFile.GetFileFromApplicationUriAsync(uri);
+            var localYamlDefinitionsFile = await FileIO.ReadTextAsync(storageFile);
+            var localYamlDefinitions = BuildYamlDeserializer().Deserialize<List<DistributionDefinition>>(localYamlDefinitionsFile);
+            foreach (var localYamlDefinition in localYamlDefinitions)
+            {
+                // Ignore distributions that we have in the local yaml file but are no longer present in the web file.
+                if (!distributionDefinitionsMap.TryGetValue(localYamlDefinition.Name, out var definitionFromWeb))
+                {
+                    continue;
+                }
+
+                definitionFromWeb.Publisher = localYamlDefinition.Publisher;
+                definitionFromWeb.WindowsTerminalProfileGuid = localYamlDefinition.WindowsTerminalProfileGuid;
+
+                // Only add a logo to the definition we got from the web if the definition in the local yaml file
+                // has one.
+                if (!string.IsNullOrEmpty(localYamlDefinition.LogoFile))
+                {
+                    // Update the logo with the base64 string representation so we can show it as a thumbnail.
+                    var logoFilePath = WslLogoPathFormat.FormatArgs(localYamlDefinition.LogoFile);
+                    definitionFromWeb.Base64StringLogo = await _packageHelper.GetBase64StringFromLogoPathAsync(logoFilePath);
+                }
+            }
+
+            return distributionDefinitionsMap;
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, "Unable to retrieve all definitions for known distributions");
+            return new Dictionary<string, DistributionDefinition>();
+        }
+    }
+
+    private bool ShouldAddDistribution(DistributionDefinition distribution)
+    {
+        if (_osArchitecture == Architecture.Arm64)
+        {
+            return distribution.IsArm64Supported;
+        }
+        else if (_osArchitecture == Architecture.X64)
+        {
+            return distribution.IsAmd64Supported;
+        }
+
+        return false;

```

Comment:
Why no check for x86?  Is it not supported?

Reply:
going off of https://aka.ms/distributionsinfojson we only allowing 64 bit versions since these are the ones we know about

Reply:
WSL only works on 64bits